tests/sources/test_zoom.py

```diff
@@ -0,0 +1,801 @@
+    "type": "file",
+    "date_time": "2023-02-09T00:00:00Z",
+    "file_size": 10485761,
+    "file_name": "file5.txt",
+    "download_url": "https://api.zoom.us/v2/download_url",
+}
+FILE_WITH_UNSUPPORTED_EXTENSION = {
+    "file_id": "file4",
+    "type": "file",
+    "date_time": "2023-02-09T00:00:00Z",
+    "file_size": 300,
+    "file_name": "file3.png",
+    "download_url": "https://api.zoom.us/v2/download_url",
+}
+SAMPLE_FILE = {
+    "next_page_token": None,
+    "messages": [
+        FILE,
+        FILE_WITHOUT_EXTENSION,
+        FILE_WITH_LARGE_DATA,
+        FILE_WITH_UNSUPPORTED_EXTENSION,
+    ],
+}
+EXPECTED_FILES_RESPONSE = [
+    {
+        "file_id": "file1",
+        "type": "file",
+        "date_time": "2023-03-09T00:00:00Z",
+        "file_size": 100,
+        "file_name": "file1.txt",
+        "download_url": "https://api.zoom.us/v2/download_url",
+        "id": "file1",
+        "_id": "file1",
+        "_timestamp": "2023-03-09T00:00:00Z",
+    },
+    {
+        "file_id": "file2",
+        "type": "file",
+        "date_time": "2023-02-09T00:00:00Z",
+        "file_size": 300,
+        "file_name": "file4",
+        "download_url": "https://api.zoom.us/v2/download_url",
+        "id": "file2",
+        "_id": "file2",
+        "_timestamp": "2023-02-09T00:00:00Z",
+    },
+    {
+        "file_id": "file3",
+        "type": "file",
+        "date_time": "2023-02-09T00:00:00Z",
+        "file_size": 10485761,
+        "file_name": "file5.txt",
+        "download_url": "https://api.zoom.us/v2/download_url",
+        "id": "file3",
+        "_id": "file3",
+        "_timestamp": "2023-02-09T00:00:00Z",
+    },
+    {
+        "file_id": "file4",
+        "type": "file",
+        "date_time": "2023-02-09T00:00:00Z",
+        "file_size": 300,
+        "file_name": "file3.png",
+        "download_url": "https://api.zoom.us/v2/download_url",
+        "id": "file4",
+        "_id": "file4",
+        "_timestamp": "2023-02-09T00:00:00Z",
+    },
+]
+
+
+class ZoomAsyncMock(mock.AsyncMock):
+    def __init__(self, data, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._data = data
+
+    async def json(self):
+        return self._data
+
+    async def text(self):
+        return self._data
+
+
+def get_mock(mock_response):
+    async_mock = mock.AsyncMock()
+    async_mock.__aenter__ = mock.AsyncMock(
+        return_value=ZoomAsyncMock(data=mock_response)
+    )
+    return async_mock
+
+
+@asynccontextmanager
+async def create_zoom_source(fetch_past_meeting_details=False):
+    async with create_source(ZoomDataSource) as source:
+        source.configuration.get_field("account_id").value = "123"
+        source.configuration.get_field("client_id").value = "id@123"
+        source.configuration.get_field("client_secret").value = "secret#123"
+        source.configuration.get_field(
+            "fetch_past_meeting_details"
+        ).value = fetch_past_meeting_details
+        yield source
```

Comment:
```suggestion
@asynccontextmanager
async def create_zoom_source(fetch_past_meeting_details=False):
    async with create_source(
        ZoomDataSource,
        account_id="123",
        client_id="id@123",
        client_secret="secret#123",
        fetch_past_meeting_details=fetch_past_meeting_details
    ) as source:
        yield source
```

You can create the data source with the values already instead of applying them after creation.